synapseclient/api/wiki_service.py

```diff
@@ -0,0 +1,456 @@
+                instance from the Synapse class constructor.
+
+    Returns:
+        The the list of FileHandles for all file attachments of a specific WikiPage for a given owning Entity.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    # Add version as a query parameter if provided
+    params = {}
+    if wiki_version is not None:
+        params["wikiVersion"] = wiki_version
+
+    return await client.rest_get_async(
+        uri=f"/entity/{owner_id}/wiki2/{wiki_id}/attachmenthandles",
+        params=params,
+    )
+
+
+async def get_attachment_url(
+    owner_id: str,
+    wiki_id: str,
+    file_name: str,
+    *,
+    redirect: Optional[bool] = False,
+    wiki_version: Optional[int] = None,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """Get the URL of a wiki page attachment.
+    <https://rest-docs.synapse.org/rest/GET/entity/ownerId/wiki2/wikiId/attachment.html>
+
+    Arguments:
+        owner_id: The ID of the owner entity.
+        wiki_id: The ID of the wiki.
+        file_name: The name of the file to get.
+                The file names can be found in the FileHandles from the GET /entity/{ownerId}/wiki/{wikiId}/attachmenthandles method.
+        redirect: When set to false, the URL will be returned as text/plain instead of redirecting. Default is False.
+        wiki_version: Optional version of the wiki page.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The URL that can be used to download a file for a given WikiPage file attachment.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    # Add version and redirect as a query parameter if provided
+    params = {}
+    params["fileName"] = file_name
+    if wiki_version is not None:
+        params["wikiVersion"] = wiki_version
+    if redirect is not None:
+        params["redirect"] = redirect
+
+    return await client.rest_get_async(
+        uri=f"/entity/{owner_id}/wiki2/{wiki_id}/attachment",
+        params=params,
+    )
+
+
+async def get_attachment_preview_url(
+    owner_id: str,
+    wiki_id: str,
+    file_name: str,
+    redirect: Optional[bool] = False,
+    wiki_version: Optional[int] = None,
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """Get the preview of a wiki page attachment.
+    <https://rest-docs.synapse.org/rest/GET/entity/ownerId/wiki2/wikiId/attachmentpreview.html>
+
+    Arguments:
+        owner_id: The ID of the owner entity.
+        wiki_id: The ID of the wiki.
+        file_name: The name of the file to get.
+                The file names can be found in the FileHandles from the GET /entity/{ownerId}/wiki/{wikiId}/attachmenthandles method.
```

Comment:
Rather than mention what endpoint would be used here, the better method would be to mention the function one would use. This example format https://github.com/Sage-Bionetworks/synapsePythonClient/blob/d9998ef905aa98a40ed98a5590df4bd314c51ccb/synapseclient/models/entityview.py#L140 shows how the mkdocs can automatically create a hyperlink to the function when it gets rendered on the docs site